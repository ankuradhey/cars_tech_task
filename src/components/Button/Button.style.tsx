import styled from "styled-components";

export enum VARIANT {
    PRIMARY = "primary",
    SECONDARY = "secodary",
}
interface IProps {
    variant?: VARIANT;
}

export const Button = styled.button<IProps>`
    border-radius: ${(props) => props.theme.borderRadius};
    border: none;
    width: 128px;
    height: 32px;
    :focus {
        outline: none;
    }
    :active {
        color: ${(props) => props.theme.palette.secondary.contrastText};
        background-color: ${(props) => props.theme.palette.secondary.main};
    }
    :hover {
        color: ${(props) => props.theme.palette.secondary.contrastText};
        background-color: ${(props) => props.theme.palette.secondary.main};
    }
    ${(props) => {
        switch (props.variant) {
            case VARIANT.SECONDARY:
                return `
          color: ${props.theme.palette.secondary.contrastText};
          background-color: ${props.theme.palette.secondary.main};
        `;
            case VARIANT.PRIMARY:
            default:
                return `
          color: ${props.theme.palette.primary.contrastText};
          background-color: ${props.theme.palette.primary.main};
        `;
        }
    }}
`;
