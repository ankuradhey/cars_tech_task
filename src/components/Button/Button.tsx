import React, { FC } from "react";
import * as Styled from "./Button.style";

export const Button: FC = (props) => {
    return <Styled.Button {...props} />;
};
