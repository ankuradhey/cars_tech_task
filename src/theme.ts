import { DefaultTheme } from "styled-components";

export const defaultTheme: DefaultTheme = {
    borderRadius: "2px",
    palette: {
        common: {
            black: "#4A4A4A",
            white: "#EDEDED",
        },
        primary: {
            main: "#EA7F28",
            contrastText: "#ffffff",
        },
        secondary: {
            main: "#D37324",
            contrastText: "#ffffff",
        },
        tertiary: {
            main: "#ffffff",
            contrastText: "#EDEDED",
        },
    },
    gutterSize: {
        sm: "8px",
        md: "12px",
        lg: "24px",
    },
};
