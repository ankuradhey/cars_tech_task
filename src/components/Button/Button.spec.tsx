import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../../theme";
import { Button } from "./Button";

describe("Button Component", () => {
    it("Should render Button component", () => {
        const { container, getByText } = render(
            <ThemeProvider theme={defaultTheme}>
                <Button>Test me!</Button>
            </ThemeProvider>
        );
        expect(getByText("Test me!")).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });
});
