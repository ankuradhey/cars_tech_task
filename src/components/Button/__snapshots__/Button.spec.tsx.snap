// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should render Button component 1`] = `
<button
  class="sc-bdfBwQ hfyqFt"
>
  Test me!
</button>
`;
